bin/fin

```diff
@@ -3231,16 +3231,57 @@ install_proxy_service ()
 		"${IMAGE_VHOST_PROXY}" >/dev/null
 }
 
+# Detects upstream DNS or falls back to DOCKSAL_DEFAULT_DNS
+# If DOCKSAL_DNS_UPSTREAM is preset, then skips DNS detection
+detect_dns ()
+{
+	# Do not detect, if overridden in docksal.env
+	[[ "$DOCKSAL_DNS_UPSTREAM" != "" ]] && return
+
+	if is_linux || is_mac; then
+		# Get from /etc/resolv.conf
+		nameservers=$(grep "nameserver" "/etc/resolv.conf" 2>/dev/null | sed "s/nameserver //")
+		# Find first suitable
+		if [[ "$nameservers" != "" ]]; then
+			for server in $nameservers; do
+				# Check that it is an IP address and exclude Docksal IP
+				if [[ "$server" =~ [0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3} ]] &&
+					[[ "$server" != "192.168.64.100" ]]; then
+					DOCKSAL_DNS_UPSTREAM="$server"
+				fi
+			done
+		else
+			# In case no name servers found, fall back to the default
+			DOCKSAL_DNS_UPSTREAM="$DOCKSAL_DEFAULT_DNS"
+		fi
+	elif is_windows; then
+		# Look for Primary WINS server, it represents DNS whe WINS services is in use
+		local primary_wins=$(ipconfig /all | grep "Primary WINS")
+		if [[ "$primary_wins" =~ ([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}) ]]; then
+			DOCKSAL_DNS_UPSTREAM=${BASH_REMATCH[0]}
```

Comment:
OMG what does that even do?? :)

Reply:
BASH_REMATCH array contains matches from the last regexp matching (line above).